### ISSUE TITLE
[Enhancement] Add SASL support to Topic Operator for standalone use

### ISSUE THREAD

**Issue by Coqueiro** [strimzi/strimzi-kafka-operator#3741]
I've followed the [documentation on how to deploy a standalone Topic Operator](https://strimzi.io/docs/operators/master/deploying.html#deploying-the-topic-operator-standalone-str).

It asks for those variables:
```
STRIMZI_TRUSTSTORE_LOCATION
STRIMZI_TRUSTSTORE_PASSWORD
STRIMZI_KEYSTORE_LOCATION
STRIMZI_KEYSTORE_PASSWORD
```

That said, we have a MSK cluster using SCRAM over SASL_SSL (https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html), we use signed certificates, so we don't need a store.
We also started to work with Confluent Cloud, doing PLAIN over SASL_SSL, also signed certificates.

Is there a way to setup the Topic Operator over those specific scenarios?

**Comment by ppatierno** [comment]
@scholzj I did a research on issues but even taking a look at the code, we don't have SASL support in the Topic Operator.

**Comment by Coqueiro** [comment]
Since Confluent and MSK already support those authentications methods, this would be a great addition.

**Comment by scholzj** [comment]
If you want, we can convert this to enhancement. But I will be honest with you, from the core maintainers our main focus is to have the ecosystem working around the cluster operator and our own Kafka and not around Amazon MSK. So you should probably not expect this to get implemented any time soon unless someone contributes it. (we can of course provide help and guidance during the implementation)

**Comment by Coqueiro** [comment]
Totally understandable reasoning. Thanks for the clarity, @scholzj !

Would be great to use Strimzi's Topic and User Operator in our current setup, they are good solutions from a DevOps perspective. Sooner of later I may try to implement this.

**Comment by AlbertoPeon** [comment]
Hi @ppatierno,

Does the same apply to the Kafka User operator? We were planning to use both with MSK and SASL_SSL (we have a very similar usecase as OP) but this is a bit of a blocker for us.

**Comment by ppatierno** [comment]
Yes that's the same

**Comment by neuromantik33** [comment]
Would love to use strimzi's connect operator to also be able to connect to confluent's managed kafka cluster using plain SASL_SSL. Of course we plan to be running our own kafka eventually once we get the necessary hardware but in the meantime it would be great to be able to use strimzi for connecting to _foreign_ clusters.

**Comment by scholzj** [comment]
@neuromantik33 With _strimzi's connect operator_ you mean to run Kafka Connect? That is done by the Cluster Operator - you should be able to connect that to any Kafka broker with SASL_SSL.

**Comment by scholzj** [comment]
This should be now fixed in #5201
